client/tests/integration/components/ui/modal-test.js

```diff
@@ -0,0 +1,30 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | ui/modal', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('it opens', async function(assert) {
+    this.open = false;
+    this.toggle = () => { this.set('open', !this.open); };
+
+    // Template block usage:
+    await render(hbs`
+      <div id="reveal-modal-container">
+      </div>
+
+      <Ui::Modal
+        @open={{this.open}}
+        @toggle={{this.toggle}}
+      >
+        template block text
+      </Ui::Modal>
+    `);
+
+    await this.toggle();
+
+    assert.dom().containsText('template block text');
```

Comment:
Nit picky thing: maybe you should also include a negative test before you run `await this.toggle()`

e.g. `assert.dom()doesNotContainText('template block text')` so that you know the modal is in fact opening when you toggle it (and not open by default or opening for some other reason)

Reply:
Aah! Yep. Actually... I bet you the negative test fails. Will fix.